"Biscuit" cards, and also now slip papers are actually paper!

DrDiasyl opened the pull request:
## About The Pull Request
**This is not a literal biscuit which you eat. "Biscuit" is just a nickname of this kind of snap-cards.**
Slip paper is now actually paper, before that, it was just literally nothing, an object which does nothing. Because of that, the paper cutter was useless as well, as it turned normal paper into useless paper slips. 
Because of this now paper cutters are placed at:

- Bridge
- HoP office
- Warden office
- Cargo

Also adds biscuit cards. If you have no idea what it is, it's a kind of card placed around a piece of paper usually to conceal some document and it can be opened by cracking it with a crunchy sound (that's why it's named a biscuit). Those are usually only opened in certain situations or emergencies (The most famous example is the US president's nuclear biscuit, which contains nuclear codes)

There are 2 biscuits: Normal and Confidential.
Normal is a plain biscuit with nothing really special, can be coloured if you want as it's white.
![image](https://user-images.githubusercontent.com/42353186/229195802-4ac5ad96-6083-47dc-9ccd-13d132f690bf.png)

The confidential biscuit is blue and has "NT" on it.
![image](https://user-images.githubusercontent.com/42353186/229195753-207d64f1-1f17-418e-a01a-8bfa2c4abc51.png)

There is a not-sealed biscuit you can print in autolathe, it starts open so you can put paper in and seal later making it a normal crackable one.

Now spare ID safe code automatically spawns in a confidential biscuit in a new paper slip subtype - corporate slip paper. It's a plastic card which is sturdier than the normal paper clip.
![image](https://user-images.githubusercontent.com/42353186/229195696-07075f78-c736-43a9-bd14-5cf9ca36ddc9.png)

You can also create the corporate paper slip paper using - normal paper slip, plastic, and captain rubber stamp (making this paper having a rare status)

![image](https://user-images.githubusercontent.com/42353186/229195599-75807483-02f9-44cb-ae24-d3388b509ccd.png)
![image](https://user-images.githubusercontent.com/42353186/229195610-5a264ed4-8b04-4452-9032-5755e2419423.png)


## Why It's Good For The Game
Paper slips being not paper is stupid and has no point in being that way. This also made the paper cutter useless; hence, it was never placed on any map (and never used).
This PR makes it paper, so it's usable when crew members need a paper slip instead of a full-size sheet.

Placing paper cutters around the station makes it so the crew can produce paper slips when they need them, this also gives it finally a use and place on the station.

The 'biscuits' are cool and give more flavour to the spare ID safe codes, as well as to other documents people might put in future PRs. This also makes those paranoid captains which are scared of their spare ID safe code being stolen and read can sleep tight, as now if someone opens it up it's really obvious. Now also you can really make sure if that head is lying about "not touching the spare ID", _but not as anyone cares really._

**Okay, it's draft for now just so I can add some new things, and fix the dupe bug:**

- [x] Being able to fax the paper biscuits
- [x] Make it impossible to cut paper slips into paper slips
- [x] Make corporate paper slips craftable or printable
- [x] Make confidential biscuits craftable or printable
- [x] Make paper cutters orderable at the cargo

_CRUNCH_
## Changelog
:cl: DrDiasyl aka DrTuxedo#0931
add: Added 'biscuit' cards! They can contain documents and can only be accessed by cracking them open, you can't close them back. Nanotrasen now stores spare ID safe codes in them.
add: Placed paper cutters around the station. They're in Bridge, HoP office, Warden office, and Cargo.
add: Now you can order paper cutters at cargo.
fix: Now the paper slip is actually paper.
imageadd: The paper slips sprite was slightly tweaked to have text lay more logically, added the corporate paper slip.
/:cl:

DrDiasyl committed:
Adds paper biscuits and makes paper slips usable

Now paper slips are paper and can be written on. Added subtype of paper slip which is fancier "corporate paper slip".
Now there is a biscuit card, like the one used for the nuclear code of the US president. It's a folder which can only be opened when cracked, cracking is obvious and irreversible
Now spare ID codes spawn in biscuits.

DrDiasyl committed:
Places papercutters around stations

Now there is a way to make paper slips on the station.
Every station has a paper cutter at Bridge, HoP office, Warden office and Cargo.

Also deletes singular paper slip from maintenance at Metastation.

DrDiasyl commented:
**code/modules/paperwork/paper_cutter.dm** conflicting? But with who? I don't see any PR which touches the paper_cutter.dm (Who cares about it?)

Jacquerel commented:
It's conflicting with #74202 because your branch is old and out of date, the same reason there are conflicts in map files.
Fix the code conflict through whatever your preferred git workflow is but make sure to install the map githooks before pulling and it might be able to automatically fix that for you (otherwise it's not too hard to fix in sdmm, look for merge conflict markers)

DATA-xPUNGED commented:
Is there a way to craft these?

DrDiasyl commented:
> Is there a way to craft these?

Actually right now there is no way to craft these. While I fix the merge conflict I can make them craftable.

DrDiasyl commented:
> It's conflicting with #74202 because your branch is old and out of date, the same reason there are conflicts in map files.
> Fix the code conflict through whatever your preferred git workflow is but make sure to install the map githooks before pulling and it might be able to automatically fix that for you (otherwise it's not too hard to fix in sdmm, look for merge conflict markers)

_How you do this?_

Wallemations commented:
It being named a "biscuit card" made me want these to be used in fortune cookie recipes.
I'm genuinely surprised we don't have fortune cookies.

DrDiasyl commented:
> It being named a "biscuit card" made me want these to be used in fortune cookie recipes.
> I'm genuinely surprised we don't have fortune cookies.

There is fortune cookies in game.

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+	else ..()
+
+/obj/item/folder/biscuit/attack_self(mob/user)
+	add_fingerprint(usr)
```
```suggestion
	add_fingerprint(user)
```

no need for `usr` memes here

DrDiasyl replied:
It was that way in the normal paper code where I copied it LMAO, I didn't do this on intention. But alright, might even change usr to user in the paper code.

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+
+//All next is done so you can't reach contents, or put any new contents when its not cracked open
+/obj/item/folder/biscuit/remove_item(obj/item/Item, mob/user)
+    if (crack_check(user))
+        return ..()
```
```suggestion
    if (!crack_check(user))
        return
    return ..()
```

prefer to do it like this

DrDiasyl replied:
This broke it too

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+    return FALSE
+
+//All next is done so you can't reach contents, or put any new contents when its not cracked open
+/obj/item/folder/biscuit/remove_item(obj/item/Item, mob/user)
```
```suggestion
/obj/item/folder/biscuit/remove_item(obj/item/item, mob/user)
```

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+        return ..()
+
+/obj/item/folder/biscuit/attack_hand(mob/user, list/modifiers)
+	if (LAZYACCESS(modifiers, RIGHT_CLICK))
+		if (crack_check(user))
+			return ..()
+	else ..()
```
```suggestion
	if (LAZYACCESS(modifiers, RIGHT_CLICK) && crack_check(user)
			return ..()
	return
```

DrDiasyl replied:
@san7890 this **broke** it! Are you sure this is really necessary?

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+	else ..()
+
+/obj/item/folder/biscuit/attackby(obj/item/weapon, mob/user, params)
+	if(is_type_in_typecache(weapon, folder_insertables))
+		if (crack_check(user))
+			return ..()
+	else ..()
```
```suggestion
	if(is_type_in_typecache(weapon, folder_insertables) && crack_check(user))
		return ..()
	
	return
```

DrDiasyl replied:
This broke it too

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+			icon_state = "[icon_state]_cracked"
+			update_appearance()
+		return
+	if (cracked)
+		ui_interact(user)
+	return
```
```suggestion
	if (!cracked)
		return

	ui_interact(user)
```

doing it like this makes that return at the end seem not-so-superfluous

DrDiasyl replied:
This broke it too

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+			cracked = TRUE
+			playsound(get_turf(user), 'sound/effects/wounds/crack1.ogg', 70, TRUE)
+			icon_state = "[icon_state]_cracked"
+			update_appearance()
+		return
```
```suggestion
			update_appearance()
		else
			return
```

i have a feeling this is what you were going for

DrDiasyl replied:
Will the code still return if you crack it then? This won't break anything? @san7890

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+	desc = "An biscuit card"
+	icon_state = "paperbiscuit"
+	bg_color = "#b88f3d"
+	var/cracked = FALSE //Is biscuit cracked open or not?
```
```suggestion
	/// Is biscuit cracked open or not?
	var/cracked = FALSE
```

Do it like this (with `///` above the variable/proc) to DMDoc it.

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+	if(length(contents))
+		. += span_notice("To reach contents you need to crack it open.")
+
+/obj/item/folder/biscuit/proc/crack_check(mob/user)
```
DMDoc this, I showed you how above

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,80 @@
+/obj/item/folder/biscuit/proc/crack_check(mob/user)
+    if (cracked)
+        return TRUE
+    balloon_alert(user, "need to crack it open first!")
```
```suggestion
    balloon_alert(user, "unopened!")
```

balloon_alerts should be more brief, this carries the point imo, but you can do whatever you want

san7890 reviewed (changes requested):
i see why you did what you did but there's much better ways to accomplish it (`else ..()` is pretty awk)

DrDiasyl commented:
Okay, as this was turned into a draft I think I will make some tweaks while fixing merge conflicts and reviews.
I want to make corporate paper slips made out of plastic cards and be a "card".
Also, I slightly changed the biscuit sprite so it will look more cracked then torn apart, maybe also change it from brown cardboard colour into a plastic one as well.

DrDiasyl committed:
New sprites, code changes to review

Changed the code to match the review, but with some tweaking or else it won't work.
Biscuits got prettier sprites.
Now there is a pre-open biscuit you can print in autolathe using plastic, you can put paper in it and seal it to work as a normal biscuit.

Also, an attempt to make a paper cutter not cut paper slips, as that is a literal dupe. But yet it's not successful.

DrDiasyl committed:
Revert "Places papercutters around stations"

This reverts commit fe83716ee7a7409af9fb4e85f00087e32f7938b3.

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into slip-paper-fix

DrDiasyl committed:
Did I just fix merge conflict of paper cutter?

DrDiasyl committed:
Places paper cutters around map (again)

Lets hope this time its merge conflict free

DrDiasyl commented:
> Is there a way to craft these?

Now you can!
As those are now **printable on autolathe** with plastic!

DrDiasyl commented:
Oh wait, why I removes draft from it too soon?! Dummy me, you still can somehow dupe paperslips by cutting paperslips into paperslips.
Although this seems like not the hardest thing to fix, at least _I guess_. If you can help - help me.

DrDiasyl commented:
FUCKING **INDENTIONS**

DrDiasyl committed:
...fixes indents?

DrDiasyl commented:
Okay this was a mistake, never trust github editor on phone.

DrDiasyl committed:
maybe now?

DrDiasyl committed:
forgot about this part

fixed indention in the paper cutter blade, no idea why I touched it bruh

DrDiasyl commented:
Okay whatever, will fix it when will get onto PC

kinneb commented:
> Wait what? aren't they biscuits? I tought they were considered grain?

read the pr body they arent actual biscuits

Jezzared commented:
Can these be run through the fax?

Fax the Safe Code card to some random fax on the station and tell the assistants to go wild.

Also, might make sense for the paper slips to have a lower character limit than regular paper, since they are smaller.

DrDiasyl commented:
> Can these be run through the fax?
> 
> Fax the Safe Code card to some random fax on the station and tell the assistants to go wild.
> 
> Also, might make sense for the paper slips to have a lower character limit than regular paper, since they are smaller.

Good idea to make them faxable.
And about the paper slip character limit, if I find where it lays then I will change it.

DrDiasyl commented:
Also just wanna say that paper slips are already faxable as they're a subtype of paper, but biscuits are not yet.

DrDiasyl committed:
fixes indention in paper_biscuit

DrDiasyl committed:
Merge branch 'master' into slip-paper-fix

DrDiasyl committed:
Adds into desc of biscuit to "NOT DIGEST"

Funny humour about it being named a biscuit, and also the suicide message being that you eat it.

DrDiasyl committed:
Minor fixes of commentary for clarity

DrDiasyl committed:
Makes paper biscuit bg_color white

It is literally white

DrDiasyl committed:
Merge branch 'slip-paper-fix' of https://github.com/DrDiasyl/tgstation into slip-paper-fix

DrDiasyl committed:
Revert "Places paper cutters around map (again)"

This reverts commit 299140fc6d00f6c25123c540a397c76584de805e.

DrDiasyl commented:
Why map changes are the most pesky ones?

DrDiasyl committed:
Fixes the CI Suite Map failure

DrDiasyl committed:
Places paper cutters around station (AGAIN AGAIN)

DrDiasyl committed:
You can now fax paper slips and order paper cutters

DrDiasyl committed:
now you can't cut paperslips into paperslips

DrDiasyl commented:
Okay, I decided I won't half the character limit as such will require to tamper with statics of paper and I'm scared.

DrDiasyl committed:
Makes confidental biscuit and corporate paper slip obtainable

You can print confidential biscuits on hacked autolathe.
You can make corporate paper slip with normal paper slip, plastic, and the captain's stamp.

DrDiasyl commented:
what

DrDiasyl committed:
fixed stupid typo

DrDiasyl committed:
paper biscuit integrity rise

DrDiasyl committed:
Merge branch 'master' into slip-paper-fix

DrDiasyl commented:
@san7890 what with the changes? Is this fine now?

san7890 commented on code/datums/components/crafting/misc.dm:
```diff
@@ -23,3 +23,13 @@
+	name = "Corporate Plastic Card"
+	result = /obj/item/paper/paperslip/corporate
+	time = 3 SECONDS
+	reqs = list(/obj/item/paper/paperslip = 1,
+				/obj/item/stack/sheet/plastic = 3
+	)
```
```suggestion
	reqs = list(
		/obj/item/paper/paperslip = 1,
		/obj/item/stack/sheet/plastic = 3,
	)
```

do list multilining like this, everything else looks poor (note the trailing comma)

san7890 commented on code/datums/components/crafting/misc.dm:
```diff
@@ -23,3 +23,13 @@
 	)
 	category = CAT_MISC
+
+/datum/crafting_recipe/corppaperslip
```
can you make this snake_case (e.g. `corporate_paper_slip`)

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+
+/obj/item/folder/biscuit/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!cracked)
+		if (tgui_alert(user, "Do you want to crack it open? You cannot close it back.", "Biscuit card", list("Yes", "No")) == "Yes")
+			cracked = TRUE
+			playsound(get_turf(user), 'sound/effects/wounds/crack1.ogg', 60)
+			icon_state = "[icon_state]_cracked"
+			update_appearance()
+			return
+		else
+			return
+	if (cracked)
+		ui_interact(user)
+		return
```
```suggestion
	if (!cracked)
		if (tgui_alert(user, "Do you want to crack it open? You cannot close it back.", "Biscuit card", list("Yes", "No")) == "Yes")
			cracked = TRUE
			playsound(get_turf(user), 'sound/effects/wounds/crack1.ogg', 60)
			icon_state = "[icon_state]_cracked"
			update_appearance()
		else
			return

	ui_interact(user)
```

Based on what I can gather from your code, this is probably what you want? Otherwise you'd have to use it in-hand _twice_ in order to actually access the contents and that's just weird. no need to constantly be evaluating `cracked` if we're going to be certain it's true (either if it was already cracked or if we cracked it via the tgui_alert)

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+/obj/item/folder/biscuit/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!cracked)
+		if (tgui_alert(user, "Do you want to crack it open? You cannot close it back.", "Biscuit card", list("Yes", "No")) == "Yes")
```
```suggestion
		if (tgui_alert(user, "Do you want to crack it open? You cannot uncrack it.", "Biscuit card", list("Yes", "No")) == "Yes")
```

verb consistency. you could also turn it into an early return like so:

```suggestion
		if (tgui_alert(user, "Do you want to crack it open? You cannot uncrack it.", "Biscuit card", list("Yes", "No")) != "Yes")
			return
```

so we can de-indent the rest of the code and make it look a bit neater, especially since we don't need the `else` case anymore (this just replaces the if/else with something equivalent)

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+	new /obj/item/paper/paperslip/corporate/fluff/emergency_spare_id_safe_code(src)
+
+//Biscuits which start not-sealed/cracked initially for the crafting, printing and such
+/obj/item/folder/biscuit/not_sealed
```
make this `unsealed` that's a bit more proper grammatically

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+//Biscuits which start not-sealed/cracked initially for the crafting, printing and such
+/obj/item/folder/biscuit/not_sealed
+	name = "\proper biscuit card"
+	desc = "An biscuit card. To reach contents you need to crack it open. Has label which says <b>DO NOT DIGEST</b>."
```
this desc makes no sense if it's already cracked open

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+	icon_state = "paperbiscuit_cracked"
+	///What is the sprite for when its not cracked? As it starts already cracked, and for re-sealing needs to have a sprite
+	var/not_cracked_icon = "paperbiscuit"
+	cracked = TRUE
```
vars inherited from parent must always be above new vars declared, like

```txt
cracked
var/new_var
var/new_var_2
```

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+/obj/item/folder/biscuit/not_sealed/examine()
+	. = ..()
+	if(!sealed)
+		. += span_notice("This one have never been sealed yet. Put in any contents to seal it by pressing it in hand. After sealing the only way to reach contents is by cracking it which is irreversible.")
```
```suggestion
		. += span_notice("This one have not been sealed yet. You many insert anything to seal it by pressing it in hand. Once sealed, the contents are inaccessible until cracked open (irreversible).")
```

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+/obj/item/folder/biscuit/not_sealed/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!sealed)
+		if (tgui_alert(user, "Do you want to seal it? After sealing the only way to reach the contents is by cracking the biscuit, you cannot re-seal it again after that.", "Biscuit card", list("Yes", "No")) == "Yes")
+			cracked = FALSE
+			sealed = TRUE
+			playsound(get_turf(user), 'sound/items/duct_tape_snap.ogg', 60)
+			icon_state = "[not_cracked_icon]"
+			update_appearance()
+			return
+		else
+			return
```
do the same thing i mentioned earlier with early returns and getting rid of the excess returns

san7890 commented on code/modules/research/designs/autolathe/multi-department_designs.dm:
```diff
@@ -550,3 +550,30 @@
+	build_path = /obj/item/folder/biscuit/not_sealed
+	category = list(
+		RND_CATEGORY_INITIAL,
+		RND_CATEGORY_TOOLS + RND_SUBCATEGORY_TOOLS_SERVICE
```
```suggestion
		RND_CATEGORY_TOOLS + RND_SUBCATEGORY_TOOLS_SERVICE,
```

san7890 commented on code/modules/research/designs/autolathe/multi-department_designs.dm:
```diff
@@ -550,3 +550,30 @@
+	build_path = /obj/item/folder/biscuit/not_sealed/confidental
+	category = list(
+		RND_CATEGORY_HACKED,
+		RND_CATEGORY_TOOLS + RND_SUBCATEGORY_TOOLS_SERVICE
```
```suggestion
		RND_CATEGORY_TOOLS + RND_SUBCATEGORY_TOOLS_SERVICE,
```

DrDiasyl committed:
Merge branch 'master' into slip-paper-fix

DrDiasyl committed:
Makes changes according to the reviews.

Now description is dynamic, it will tell you that the biscuit needs cracking only if it's not cracked.

Renamed "not_sealed" to "unsealed".

Added early returns to paper biscuits attack_self code.

Changed biscuit suicide to be more brutal.

Added missing "," to multi-department-designs build paths.

Made list in the corporate paper slip craft good looking.
Renamed "corppaperslip" in crafting recipe into "corporate_paper_slip"

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+/obj/item/folder/biscuit/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!cracked)
+		if (tgui_alert(user, "Do you want to crack it open? You cannot uncrack back.", "Biscuit card", list("Yes", "No")) != "Yes")
```
```suggestion
		if (tgui_alert(user, "Do you want to crack it open?", "Biscuit Cracking", list("Yes", "No")) != "Yes")
```

The second sentence is redundant, sorry if I put you onto that since we don't really need it.

san7890 commented on code/modules/paperwork/paper_biscuit.dm:
```diff
@@ -0,0 +1,125 @@
+/obj/item/folder/biscuit/unsealed/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!sealed)
+		if (tgui_alert(user, "Do you want to seal it? After sealing the only way to reach the contents is by cracking the biscuit, you cannot re-seal it again after that.", "Biscuit card", list("Yes", "No")) != "Yes")
```
```suggestion
		if (tgui_alert(user, "Do you want to seal it? You must crack it open to reach the contents again!", "Biscuit Sealing", list("Yes", "No")) != "Yes")
```

Bit more brief, and the "cracking" menu already says that it's an irreversible thing.

san7890 reviewed (changes requested):
last grammar and formatting stuff

DrDiasyl committed:
Apply suggestions from code review

Last grammar staff from san7890

Co-authored-by: san7890 <the@san7890.com>

DrDiasyl committed:
Merge branch 'master' into slip-paper-fix
